scRNA-seq/02-filtering_scRNA.Rmd

```diff
@@ -279,60 +282,33 @@ num_genes_exp <- colSums(detection_mat)
 
 Let's plot this using the same style and type of graph as above. 
 
-```{r genes_expressed_plot, live = TRUE}
+```{r genes_expressed_plot}
 ggplot(mapping = aes(x = num_genes_exp)) +
   geom_density(fill = "lightblue") +  
   xlab("Number of genes expressed") +
   theme_classic() 
 ```
 
+This plot helps us visualize the distribution of genes per cell and can help inform how we choose the cutoff.
+We want to remove cells with a low number of genes per cell, but not be too strict that we lose a large portion of cells from the dataset. 
 
-Because this is a well-studied data set, we actually have some cell type information for this data set.
-This is not how we will normally come across data, but we can take advantage of it to learn something about the distribution of the number of genes expressed. 
-
-First we will read in the metadata:
-
-```{r read_metadata}
-# read in metadata (use guess_max greater than the default to avoid parsing errors)
-sc_metadata <- readr::read_csv(metadata_file, guess_max = 10000)
-```
-
-The `cell` column here is a combination of the sample id (e.g. `10X_P4_3`) and the cell barcode.
-We can use `dplyr::left_join()` to match this up with the information we have from our table, but we will have to make a matching `cell_id` in a new data frame with the gene expression info first. 
-
-```{r}
-num_genes_exp_df <- data.frame(barcode = names(num_genes_exp),
-                               genes_exp = num_genes_exp
-                               ) %>%
-  dplyr::mutate(cell = paste0("10X_P4_3_", barcode))
-```
-
-
-
-In order to do this, we need to prepare a data frame that has our `num_genes_exp` numbers along side our cell-type labels.
-We will do this with `dplyr::left_join()` to merge count data to our metadata table.
+Let's see what happens if we only keep cells with > 500 expressed genes. 
 
-```{r join_metadata, live = TRUE}
-# Let's make a dataframe with this information
-num_genes_exp_df <- num_genes_exp_df %>%
-  dplyr::left_join(sc_metadata)
+```{r genes_expressed_cutoff, live = TRUE}
+ggplot(mapping = aes(x = num_genes_exp)) +
```

Comment:
It isn't clear from the text before this that this chunk will be a plot. It is probably worth adding either a comment or modifying the previous line to make that expectation clear. I say that mostly from the point of view of live coding, where having the references in the text (not just in a separate doc) is useful.